fireant/middleware/concurrency.py

```diff
@@ -0,0 +1,46 @@
+import abc
+from multiprocessing.pool import ThreadPool
+
+
+class BaseConcurrencyMiddleware(abc.ABC):
+    """
+    The abstract base class that should be inherited from to define a concurrency middleware.
+    """
+    @abc.abstractmethod
+    def fetch_queries_as_dataframe(self, queries, database):
+        """
+        Implementations of this method should execute the given queries on the supplied database and return the results.
+        :return: A list of the results of the executed queries.
+        """
+        pass
+
+    def fetch_query(self, query, database):
```

Comment:
is this meant to be called from outside of the child classes?

Reply:
Yes, this is already the case.